### PR TITLE
feat: Human in the loop

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary

'Human in the loop' support for Gmail and Slack

## Review / Merge checklist

https://linear.app/n8n/issue/NODE-1480/custom-approval-operations-in-nodes-that-send-messages

### CODE REVIEW

File: packages/nodes-base/utils/sendAndWait.utils.ts [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+								show: {
+									approvalType: ['single', 'double'],
+								},
+							},
+						},
+						{
+							displayName: 'Disapprove Button Label',
+							name: 'disapproveLabel',
+							type: 'string',
+							default: 'Disapprove',
```
Comment (ShireenMissi) [inline packages/nodes-base/utils/sendAndWait.utils.ts:191]: shouldn't the default of this be "Decline" instead?

File: packages/nodes-base/utils/sendAndWait.utils.ts [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+	return config;
+}
+
+// Create Email Helpers ----------------------------------------------------------------------------------
+
+function createEmailBody(message: string, buttons: string, instanceId?: string) {
+	const utm_campaign = instanceId ? `&utm_campaign=${instanceId}` : '';
+	const n8nWebsiteLink = `https://n8n.io/?utm_source=n8n-internal&utm_medium=send-and-wait${utm_campaign}`;
+	return `
+<!DOCTYPE html>
```
Comment (ShireenMissi) [inline packages/nodes-base/utils/sendAndWait.utils.ts:300]: having html code inline makes this code very hard to read, can you extract this to a different file?

**Review by ShireenMissi — COMMENTED** [review]
Tested Manually and it all works 🎉
left a few minor comments, also could you please add some tests 🙏

**Review by ShireenMissi — APPROVED** [review]
LGTM 🎉

**Comment by janober** [comment]
Got released with [`n8n@1.63.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.63.0)

**Comment by szab100** [comment]
Hi @michael-radency, @janober,

I'm currently building a custom node integration for [SwipeFlow](https://swipeflow.io/) — a workflow approval product designed specifically for human-in-the-loop use cases (currently in public beta, with features like item revisions coming soon).

I'd love to support the new sendAndWait feature, as pausing workflows is a valuable option for simpler use cases (in addition to the preferred, separate event-driven workflows approach). However, I noticed that the interfaces and utility functions used in the core human-in-the-loop nodes aren't exported via n8n-nodes-base and don't appear to be accessible to 3rd-party nodes. Even if they were, I’m unsure whether importing them would be safe due to potential version mismatches with the host n8n instance.

Is there any plan to expose these utilities for external use, or will sendAndWait remain limited to built-in nodes for now?

Thanks in advance — and great work on the feature!

**Pull request merged**